### ISSUE TITLE
Options Dialog Box Sizing and Size Persistence Issues; Resetting of Settings Files

### ISSUE THREAD

**Issue by NDeLancie** [microsoft/XmlNotepad#367]
XmlNotepad ver. 2.9.0.7 / Windows 10 Pro ver. 22H2, build 19045.3930

Chris:

**Options Dialog Box Sizing and Size Persistence Issues**;

The Options dialog box (View > Options) has the following issues:

1. While the Options dialog box can be resized and relocated once opened, its size does not persist, even within the same XmlNotepad session (the next opening of the dialog box will have its upper left corner at the location of the upper left corner of the last closed dialog box, but its size will revert to the default size).

2. While the Options dialog box can be made wider (by grabbing a vertical border), taller (by grabbing a horizontal border--note that there is no visible bottom border), but the invisible bottom border can be grabbed), or both (by grabbing the lower right hand corner resizing marker), this is a one-way operation: once made taller or wider, the dialog box cannot be made less tall or less wide (of course, given that this resizing will not stick at this point beyond the current display of the dialog box, this is not really an issue, but if no. 1 above is fixed, it may be, and worse so).

**Resetting of Settings Files**:

Also, I am finding that both my Roaming AppData XmlNotepad.settings file and %StartupPath%Resources  XmlNotepad.Template.settings file get reset to default (the former occurring more frequently than the latter) under circumstances of which I am not yet sure. I will try to determine what those circumstances are and report what I find. At this time, however, I know that this happens. My only non-default settings (other than Options dialog box size and location) are the editor and the font and font size).

**General Thanks**:

Finally, as a longtime user of the 2007 XML Notepad application, thank you, Chris, for the work you have done to reinvigorate it!

**Comment by lovettchris** [comment]
Good bug about the options dialog, thanks.  I pushed a fix that makes the option dialog remember your size (but only if it is larger than the default size because that ensures something is visible and window is not super tiny).  Then it lets you resize as much as you want.  I also fixed the preservation of the window location so it is relative to the location of the Main Window.

I also decided to remove preservation of the find dialog size because there are actually 4 different sizes to remember depending on whether you are doing find, replace, and with XPath setting checked or unchecked.  Preserving one size can make these other versions of the find dialog really ugly and or dysfunctional (info clipped).
